### PR TITLE
Disable copying dlls when creating windows package

### PR DESCRIPTION
**Pull request by mattip**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Replaces #50, keeping the cleanup and `sys.version` changes, but now should not copy any dlls into `lib_pypy`, rather they should be taken from the conda environment.

### CODE REVIEW

File: recipe/patches/0018-Disable-copying-dlls.patch [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+-                tktcldir = p.dirpath().join('..').join('lib')
+-                shutil.copytree(str(tktcldir), str(pypydir.join('tcl')))
+-            except WindowsError:
+-                print("Packaging Tk runtime failed. tk86t.dll and tcl86t.dll "
+-                      "found in %s, expecting to find runtime in %s directory "
+-                      "next to the dlls, as per build "
+-                      "instructions." %(p, tktcldir), file=sys.stderr)
+-                import traceback;traceback.print_exc()
+-                raise MissingDependenciesError('Tk runtime')
+-
```
Comment (mattip) [inline recipe/patches/0018-Disable-copying-dlls.patch:70]: Hmm, maybe this part is needed? Does the conda tkinter support know where to find the runtime files?

**Comment by mattip** [comment]
It seems to work, and prints 
```
Python 3.7.10 | packaged by conda-forge | (77787b8f, Jul 28 2021, 14:32:04)
[PyPy 7.3.5 with MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
```
Importing ssl and tkinter on windows works as well. One small niggle: without activating a conda environment, I can run CPython's python.exe, but PyPy's python.exe needs support DLLs that are only available after activating the environment. Is that something I should fix? For instance without activating cython I can run this
```
d:\>\miniconda\envs\cpython\python.exe
Python 3.8.10 | packaged by conda-forge | (default, May 11 2021, 06:25:23) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
```

but this complains about missing `libexpat.dll`, `LIBBZ2.dll` and `zlib.dll`.
```
d:\>\miniconda\envs\pypy37\python
```

File: recipe/patches/0018-Disable-copying-dlls.patch (hunk 1 above)
Comment (mattip) [reply recipe/patches/0018-Disable-copying-dlls.patch:70]: apparently it does.

**Comment by isuruf** [comment]
Thanks.

> For instance without activating cpython I can run this

If I get https://github.com/conda-forge/python-feedstock/pull/471 working, then cpython wouldn't too.

**Comment by mattip** [comment]
Whoot. Does this now require rerendering the open migration PRs or will it be enough to close/open the PRs to get the new version?

**Comment by h-vetinari** [comment]
Since it's "just" a build number increase, it should be picked up automatically as soon as the finished packages are through the CDN

File: recipe/patches/0001-Add-conda-forge-version.patch [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+diff -r a62de68281e3 -r 3bd4e146c802 pypy/module/sys/version.py
+--- a/pypy/module/sys/version.py	Fri Jul 02 18:18:41 2021 +0300
++++ b/pypy/module/sys/version.py	Tue Jul 27 10:00:54 2021 +0300
+@@ -55,11 +55,11 @@
+     ver = "%d.%d.%d" % (PYPY_VERSION[0], PYPY_VERSION[1], PYPY_VERSION[2])
+     if PYPY_VERSION[3] != "final":
+         ver = ver + "-%s%d" %(PYPY_VERSION[3], PYPY_VERSION[4])
+-    template = "%d.%d.%d (%s, %s, %s)\n[PyPy %s with %%s]" % (
++    template = "%d.%d.%d | packaged by conda-forge | (%s, %s, %s)\n[PyPy %s with %%s]" % (
```
Comment (tkelman) [inline recipe/patches/0001-Add-conda-forge-version.patch:17]: I think this broke package builds, see https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=357434&view=logs&jobId=856e153d-59ad-5597-92a7-c22c7f7f2082&j=856e153d-59ad-5597-92a7-c22c7f7f2082&t=2f20f7ee-86c8-5a1e-38c4-b2540d093bc9
`ValueError: failed to parse PyPy sys.version: '3.7.10 | packaged by conda-forge | (77787b8f, Jul 28 2021, 19:20:52)\n[PyPy 7.3.5 with GCC 9.3.0]'`

Comment (mattip) [reply recipe/patches/0001-Add-conda-forge-version.patch:17]: Oy oy oy. Sorry.

**Comment by mattip** [comment]
It seems more testing is needed.

File: recipe/patches/0001-Add-conda-forge-version.patch (hunk 2 above)
Comment (tkelman) [reply recipe/patches/0001-Add-conda-forge-version.patch:17]: think it'll need to match this regex https://github.com/python/cpython/blob/3.9/Lib/platform.py#L995-L998 ?

**Pull request merged**